cli/tests/unit/sleep_test.ts

```diff
@@ -0,0 +1,62 @@
+import { assert, unitTest } from "./test_util.ts";
+
+unitTest({ perms: { hrtime: true } }, function sleepSync(): void {
+  let start = performance.now();
+  Deno.sleepSync(0);
+  let after = performance.now();
+  assert(after - start >= 0);
+  assert(after - start < 1);
```

Comment:
As with setTimeout, sleep has the same caveat that the real only guarantee is that the thread slept for at-least N milliseconds.

Reply:
yup, that's why the other tests check for >= N, I found that the overhead of a simple sleep with no async shenanigans in the background was consistently under 1ms, hence the second assert